### PR TITLE
(Ready) Fixes slideshow mode

### PR DESCRIPTION
**Pull request by Kmc2000**
## About The Pull Request

(Needs testmerge)
Reduces the accuracy of projectile homing to (by my count) significantly improve projectile performance.
Turns out, homing was being called 10 odd times a second, perfoming a load of expensive calculations after the fact. This PR throttles them down to around 1 time every 0.7 seconds, and reduces the accuracy of their homing to "good enough" from "quite frankly ridiculous and unnecessary"

## Why It's Good For The Game

Lag bad, lag remove good.

## Changelog
:cl: Kmc2000
tweak: Projectiles shouldn't lag as hard during ship to ship combat
/:cl:

### CODE REVIEW

File: nsv13/code/_onclick/hud/parallax.dm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 /area/maintenance/ship_exterior //Used so that the FTL effect doesnt go away when you step outside.
 	name = "Ship exterior"
 	icon_state = "space_near"
-	dynamic_lighting = DYNAMIC_LIGHTING_IFSTARLIGHT
+	dynamic_lighting = DYNAMIC_LIGHTING_DISABLED
```
Comment (Jalleo) [inline nsv13/code/_onclick/hud/parallax.dm:10]: No just disable red alert activating at battle mode.

Comment (francinum) [reply nsv13/code/_onclick/hud/parallax.dm:10]: This should not be disabled.

File: nsv13/code/controllers/subsystem/processing/overmap.dm [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 PROCESSING_SUBSYSTEM_DEF(overmap)
 	name = "Overmap Processing"
-	wait = 2
+	wait = 1
 	stat_tag = "OP"
-	priority = FIRE_PRIORITY_OVERMAP
```
Comment (francinum) [inline nsv13/code/controllers/subsystem/processing/overmap.dm:7]: Blocks merging. Magic number bad.

File: nsv13/code/controllers/subsystem/processing/overmap.dm [hunk 3]
```diff
@@ -2,7 +2,7 @@
...
+	priority = 10
```
Comment (Jalleo) [inline nsv13/code/controllers/subsystem/processing/overmap.dm:7]: confirmed CHANGE THE DAMN DEFINE!

Comment (Kmc2000) [reply nsv13/code/controllers/subsystem/processing/overmap.dm:7]: testing only

File: nsv13/code/modules/overmap/camera.dm [hunk 4]
```diff
@@ -65,6 +65,9 @@
 		M.client.check_view()
 	var/mob/camera/aiEye/remote/overmap_observer/eyeobj = M.remote_control
 	M.cancel_camera()
+	if(M.client) //Reset px, y
```
Comment (Jalleo) [inline nsv13/code/modules/overmap/camera.dm:68]: why is this changed just asking

Comment (Kmc2000) [reply nsv13/code/modules/overmap/camera.dm:68]: It resolves a weird issue where your screen would be skewed / offset when stopping piloting ships.

File: code/modules/projectiles/projectile.dm [hunk 5]
```diff
@@ -502,7 +504,7 @@
 		var/matrix/M = new
 		M.Turn(Angle)
 		transform = M
-	if(homing)
+	if(homing && world.time >= next_homing_process) //Nsv13 performance enhancements
```
Comment (Jalleo) [inline code/modules/projectiles/projectile.dm:507]: you might need a (world.time >= next_homing_process)) not sure though

File: code/modules/projectiles/projectile.dm [hunk 6]
```diff
@@ -534,13 +536,12 @@
 		animate(src, pixel_x = trajectory.return_px(), pixel_y = trajectory.return_py(), time = 1, flags = ANIMATION_END_NOW)
 	Range()
 
-/obj/item/projectile/proc/process_homing()			//may need speeding up in the future performance wise.
-	if(!homing_target)
+/obj/item/projectile/proc/process_homing() //Nsv13 - Enhanced the performance of this entire proc.
```
Comment (Jalleo) [inline code/modules/projectiles/projectile.dm:539]: Confirm NOTHING BUT YOUR CODE USES THIS PROC FIRST PLEASE!

File: nsv13/code/__HELPERS/priority_announce.dm [hunk 7]
```diff
@@ -17,6 +17,5 @@
 		if(AR.redalert == state)
 			continue
 		AR.redalert = state
-		spawn(0) //We're branching here so that looping through the areas doesnt depend on the lighting loop finishing. Makes me wish for multithreading :(
```
Comment (Jalleo) [inline nsv13/code/__HELPERS/priority_announce.dm:20]: Ya sure about that?

File: nsv13/code/__HELPERS/priority_announce.dm [hunk 8]
```diff
@@ -17,6 +17,5 @@
...
 		for(var/obj/machinery/light/L in AR)
-			L.update()
\ No newline at end of file
+			L.update_icon()
```
Comment (Jalleo) [inline nsv13/code/__HELPERS/priority_announce.dm:21]: Honestly a CHECK_TICK while in this proc might be a good idea

File: nsv13/code/__HELPERS/priority_announce.dm [hunk 9]
```diff
@@ -17,6 +17,5 @@
 		if(AR.redalert == state)
```
Comment (Jalleo) [inline nsv13/code/__HELPERS/priority_announce.dm:17]: above here it is checking all the teleportlocs instead I believe it should only check the station_areas which is its own list and should be "SMALLER" (The longer term thing would have to pass which list it should check down the line)

File: nsv13/code/__HELPERS/priority_announce.dm (hunk 8 above)
Comment (Kmc2000) [reply nsv13/code/__HELPERS/priority_announce.dm:21]: Potentially, but this way it's done in one very brief burst of calculation rather than being dragged out

File: nsv13/code/modules/overmap/physics.dm [hunk 10]
```diff
@@ -78,8 +79,29 @@
...
+		stoplag() //Lock up the thread for a bit, throttle the process down to whatever the server can handle right now.
+		if(last_process < world.time - 0.5 SECONDS)
+			process()
+
+/obj/structure/overmap/process()
+	set waitfor = FALSE
+	var/time = min(world.time - last_process, 10)
+	time /= 10 // fuck off deciseconds
+	if(last_process > 0 && (last_process < world.time - 1 SECONDS) && !processing_failsafe) //Alright looks like the game's shat itself. Time to engage "failsafe mode". The logic of this is that if we've not been processed for over 1 second, then ship piloting starts to become unbearable and we need to step in and do our own processing, until the game's back on its feet again.
+		start_failsafe_processing()
```
Comment (Jalleo) [inline nsv13/code/modules/overmap/physics.dm:104]: ..... This should be in a specific OVERMAP PROCESSING SUBSYSTEM. NOT PER DAMN OBJECT!

File: code/modules/projectiles/projectile.dm (hunk 5 above)
Comment (Kmc2000) [reply code/modules/projectiles/projectile.dm:507]: Those evaluate the exact same

File: nsv13/code/modules/overmap/physics.dm (hunk 10 above)
Comment (Kmc2000) [reply nsv13/code/modules/overmap/physics.dm:104]: This misses the entire point of this change.

File: nsv13/code/_onclick/hud/parallax.dm (hunk 1 above)
Comment (Kmc2000) [inline nsv13/code/_onclick/hud/parallax.dm:10]:
```suggestion
	dynamic_lighting = DYNAMIC_LIGHTING_IFSTARLIGHT
```

File: nsv13/code/controllers/subsystem/processing/overmap.dm (hunk 3 above)
Comment (Kmc2000) [inline nsv13/code/controllers/subsystem/processing/overmap.dm:7]:
```suggestion
	priority = FIRE_PRIORITY_VIS
```

File: nsv13/code/modules/overmap/overmap.dm [hunk 11]
```diff
@@ -181,6 +183,7 @@
 			cabin_air.gases[/datum/gas/oxygen][MOLES] = O2STANDARD*cabin_air.volume/(R_IDEAL_GAS_EQUATION*cabin_air.temperature)
 			cabin_air.gases[/datum/gas/nitrogen][MOLES] = N2STANDARD*cabin_air.volume/(R_IDEAL_GAS_EQUATION*cabin_air.temperature)
 			move_by_mouse = TRUE //You'll want this. Trust.
+			inertial_dampeners = FALSE //Fighters by default let you move around smoothly. Big ships will want to start with IAS.
```
Comment (Kmc2000) [inline nsv13/code/modules/overmap/overmap.dm:186]:
```suggestion
```
